File: Java-Kotlin/src/main/java/offer/_53.java
package offer;

public class _53 {

    public int missingNumber(int[] nums) {
        int n = nums.length + 1;
        int last = n - 1;
        int expected = last * n / 2;
        int actual = 0;
        for (int i : nums) {
            actual += i;
        }
        return expected - actual;
    }
}

File: Go-Solutions/jianzhi/53.go
package jianzhi

func missingNumber(nums []int) int {
	n := len(nums) + 1
	last := n - 1
	expected := last * n / 2
	actual := 0
	for _, value := range nums {
		actual += value
	}
	return expected - actual
}

func missingNumber1(nums []int) int {
	i, j := 0, len(nums) - 1
	for i <= j {
		m := (i + j) / 2
		if nums[m] == m {
			i = m + 1
		} else {
			j = m - 1
		}
	}
	return i
}
